django/test/utils.py

```diff
@@ -841,6 +843,43 @@ def disable(self):
         setattr(Options, 'default_apps', self.old_apps)
 
 
+class time_keeper:
+    def __init__(self, n_tests):
+        self.test_suite_records = {}
+        self.records = []
+        self.n_tests = n_tests
+
+    def append(self, record, record_time):
+        if record[:4] == 'test':
+            self.test_suite_records[record] = record_time
+        else:
+            new_record = '%s took %.3fs' % (record, record_time)
+            self.records.append(new_record)
+
+    @contextmanager
+    def timed(self, name):
+        start_time = time.perf_counter()
+        try:
+            yield
+        finally:
+            end_time = time.perf_counter() - start_time
+            self.append(name, end_time)
+
+    def results(self):
+        results = []
+        if self.n_tests > 0:
+            sorted_times = sorted(self.test_suite_records.items(), reverse=True, key=lambda records: records[1])
+            for key, value in sorted_times:
+                if self.n_tests > 0:
+                    results.append("%s took %0.3fs" % (key, value))
+                    self.n_tests -= 1
+        if self.n_tests >= 0:
+            results.append("")
+            results.extend(self.records)
+            results = '\n'.join(results)
+            print(results)
```

Comment:
I would use `sys.stderr.write(msg + os.linesep)` for each result instead of `print()` with joined results.